test/dialects/dialects_test.py

```diff
@@ -22,57 +24,111 @@
 )
 
 
-@pytest.mark.parametrize("dialect,file", parse_success_examples)
-def test__dialect__base_file_parse(dialect, file):
-    """For given test examples, check successful parsing."""
-    raw = load_file(dialect, file)
+def _lex_and_parse(config_overrides: str, raw: str) -> Optional[BaseSegment]:
+    """Performs a Lex and Parse, with cachable inputs within fixture."""
     # Load the right dialect
-    config = FluffConfig(overrides=dict(dialect=dialect))
+    config = FluffConfig(overrides=json.loads(config_overrides))
     tokens, lex_vs = Lexer(config=config).lex(raw)
     # From just the initial parse, check we're all there
     assert "".join(token.raw for token in tokens) == raw
     # Check we don't have lexing issues
     assert not lex_vs
-
     # Do the parse WITHOUT lots of logging
     # The logs get too long here to be useful. We should use
     # specific segment tests if we want to debug logs.
-    if raw:
-        parsed = Parser(config=config).parse(tokens)
-        print(f"Post-parse structure: {parsed.to_tuple(show_raw=True)}")
-        print(f"Post-parse structure: {parsed.stringify()}")
-        # Check we're all there.
-        assert parsed.raw == raw
-        # Check that there's nothing unparsable
-        typs = parsed.type_set()
-        assert "unparsable" not in typs
+    if not raw:
+        return None
+
+    return Parser(config=config).parse(tokens)
+
+
+def lex_and_parse(config_overrides: Dict[str, Any], raw: str) -> Optional[BaseSegment]:
+    """Sets up cache parsable inputs from normal inpurts."""
+    cachable_str = json.dumps(config_overrides, sort_keys=True)
```

Comment:
Why are we converting JSON to string, only to call a function that immediately converts it back to JSON?

Reply:
Was cached in a middling step (cache_lru). I thought leaving it so designed with the doc strings wasn't a harm

Reply:
Got rid of all traces of caching